### DISCUSSION TITLE
Different render results on different video cards.

### DISCUSSION THREAD

**Discussion by demso** [MonoGame/MonoGame#8280] [Graphics]
Hello, I'm racking my brains on very strange behaviour. When my game is rendered on integrated Intel GPU I get what I expect, lighing is drawn right (first screenshot, graphic device 0 in task manager). When I choose to use Nvidia RTX 3050 discrete gpu, seem like blending is done wrong, rendering result is much darker (second screenshot, graphic device 1 in task manager). It happens when I apply gaussian blur shader.
Shader code (isDiffuse here is true): 
https://pastebin.com/guUgijGa
Method where i use this shader:
https://pastebin.com/Vz8FWAWi

I have no ideas where to start with that, if you have any i would really appreciate!
Working on laptop, project uses OpenGL.

![Скриншот 16-04-2024 011952](https://github.com/MonoGame/MonoGame/assets/41340027/7770757d-4458-43cd-a1d2-cd9266c4f27b)
![Скриншот 16-04-2024 011404](https://github.com/MonoGame/MonoGame/assets/41340027/c5869b53-14dd-4dcd-8ff5-9cb8d8ee7073)

**Comment by reitinet** [comment]
Could be a lot of things .. I would "assume" you may be just missing a ".Clear" or setting the BlendState. Maybe Integrated does some automatic things before a frame, which GPU does not.

So I would start to investigate if you are properly setting BlendStates to proper States before each Pass/DrawCall. Could be some of it just rolls over from the previous frame and then is the wrong value. Also try to isolate the issue - like only render the part in question.

**Comment by stromkos** [comment]
The `RayHandler` variable sounds like a ray-tracing kernel only available on the RTX.

Hence the light falloff.

The "blur" code would be ignored on the Intel GPU.
